### ISSUE TITLE
Refactoring request: TrySmiTag and SmiTst

### ISSUE THREAD

**Issue by paul99** [v8mips/v8mips#65]
In MIPS https://codereview.chromium.org/68793008/ (the porting of https://codereview.chromium.org/72813004), 

We worked around SmiTst() with __ And(at, value, Operand(kSmiTagMask));

Suggest we add MIPS:
    inline void SmiTst(Register value, Register scratch) {
       __ And(scratch, value, Operand(kSmiTagMask));
    }

To be used as:
    __ SmiTst(value, at);
    __ ThrowIf(ne, kNonSmiValue, at, Operand(zero_reg));

Please refactor in this patch, and elsewhere that SmiTst is used. Also consider adding NonNegativeSmiTst

Purpose is just to make code more readable, and stay similar to ARM port.

Also, please implement TrySmiTag in similar manner (it will need a scratch reg).

**Comment by kilvadyb** [comment]
Dear Paul,

I did the changes you asked but have not published yet so PTAL:
Issue 78403002: MIPS: Use SmiTst and TrySmiTag MacroAssebler instructions. - Code Review
Issue 78423002: MIPS: Improvement of “MIPS: Fixed crashes exposed though fuzzing.” (r17886). - Code Review

The second one is so small that I would add it to the first although there is difference between the purpose of them.

Best Regards,
Balazs
